src/render/tutorials/openapi-and-swagger-ui/2_0.html.md

```diff
@@ -0,0 +1,721 @@
+```
+
+`GET api/user/isadmin` invokes `admin-info` handler that returns harcoded data as either JSON or XML
+```HOCON
+{
+  operationId = admininfo-operation
+  handlers = ${config.server.handlers.common.request} [
+    {
+      name = admin-info
+    }
+  ]
+}
+```
+
+`POST /api/user/{id}/send/{message}` always returns `200 OK`
+```HOCON
+{
+  operationId = sendmsg-operation
+  handlers = [
+    {
+      name = static-status
+      config.code = 200
+    }
+  ]
+}
+```
+
+You've probably noticed our handlers' configurations reference an object called `mocked`. It's a [HOCON](https://github.com/lightbend/config/blob/master/HOCON.md) syntax and we use it just so we don't need to copy and paste hardcoded user data. Let's add the object to the file:
+
+```HOCON
+routingOperations = ${routingOperations} [
+  # ...
+]
+
+mocked = {
+  loggedUser = {
+    id = 1
+    username = "UserNr1"
+    email = "usernr1@test.test"
+    avatar = "https://api.adorable.io/avatars/285/1"
+  }
+  userList = [
+    ${mocked.loggedUser}
+    {id = 2, username = "Lorem", email = "lorem@test.test", avatar = "https://api.adorable.io/avatars/285/2"}
+    {id = 3, username = "Ipsum", email = "ipsum@test.test", avatar = "https://api.adorable.io/avatars/285/3"}
+  ]
+} 
+```
+
+### Java handlers
+
+Now let's define three handlers we used above. They're really simple.
+
+The `static-status` handler just returns a response with a given reponse code:
+
+```java
+public class StaticStatusHandlerFactory implements RoutingHandlerFactory {
+
+  @Override
+  public String getName() {
+    return "static-status";
+  }
+
+  @Override
+  public Handler<RoutingContext> create(Vertx vertx, JsonObject config) {
+    return routingContext -> routingContext
+        .response()
+        .setStatusCode(config.getInteger("code"))
+        .end();
+  }
+}
+```
+
+Nothing to explain here. The `static-json` handlers just reads its 'json` property from configuration, parses it to a JSON string and returns it:
+
+```java
+public class StaticJsonHandlerFactory implements RoutingHandlerFactory {
+
+  @Override
+  public String getName() {
+    return "static-json";
+  }
+
+  @Override
+  public Handler<RoutingContext> create(Vertx vertx, JsonObject config) {
+    return routingContext -> {
+      Object json = config.getMap().get("json");
+      routingContext.response().end(encodeJson(json));
+    };
+  }
+
+  private String encodeJson(Object object) {
+    if (object instanceof JsonObject) {
+      return ((JsonObject) object).encode();
+    } else if (object instanceof JsonArray) {
+      return ((JsonArray) object).encode();
+    } else {
+      throw new IllegalStateException("Can't encode non-json object as JSON");
+    }
+  }
+}
+```
+
+One thing that may look suspicious is the `encodeJson` method. It just calls the `encode` method on either `JsonObject` or `JsonArray` read from configuration. Its needed because, unfortunately, those classes don't share any interface with the `encode` method.

```

Comment:
Nice, one interface would be perfect here.

Reply:
The issue is hanging on vert.x github for almost 4 years :(